### ISSUE TITLE
meta.processor and meta.struct are never closed

### ISSUE THREAD

**Issue by jeff-hykin** [jeff-hykin/cpp-textmate-grammar#30]
https://github.com/atom/language-c/issues/267
```
#pragma once

#include <type_traits>

#define IsPointDef(...) \
    template<> \
    struct IsPoint<__VA_ARGS__> \
        {\
        static const bool isPoint = true;\
                }

#define ArrayBasedPointDef(T) \
    IsPointDef(T); \
    template<> \
    struct IsArrayBasedPoint<T>:public std::true_type \
        {};



#define XYBasedPointDef(T) \
    IsPointDef(T); \
    template<> \
    struct IsXYBasedPoint<T>:public std::true_type \
{};

#define TypeTAndUIsPoint \
    template<typename T, typename U, class = typename std::enable_if<IsPoint<T>::isPoint>::type, class = typename std::enable_if<IsPoint<U>::isPoint>::type>

namespace Navigation
{
    namespace Utils
    {
        template<typename T>
        struct IsPoint
        {
            static const bool isPoint = false;
        };

        template<typename T>
        struct IsArrayBasedPoint
        {
            static const bool value = false;
        };

        template<typename T>
        struct IsXYBasedPoint
        {
            static const bool value = false;
        };

    }
}
```

**Comment by matter123** [comment]
Probably blocked by https://github.com/Microsoft/vscode/issues/20488
There needs to be a way of processing multiline macro that terminates when /\\$/ no longer matches. Begin-end rules can have the nested include "overshoot" past the end, that is what is causing this.

**Comment by matter123** [comment]
The following can parse a multi-line macro-like structure
Unfortunately, it prevents using begin-end rules that cross multiple line

```json
		{
			"begin": "#define",
			"end": "(?<!\\\\)\\n",
			"name": "bold.debug",
			"contentName": "debug.debug",
			"patterns": [
				{
					"name": "single_line.debug",
					"match": "(?:\\G|^)[^\\n]+(?=\\\\$)(?<!\\\\)"
				},
				{
					"name": "single_line.debug",
					"match": "(?:\\G|^)[^\\n]+(?=$)"
				}
			]
		},
```

**Comment by jeff-hykin** [comment]
closing in favor of #136
